### PR TITLE
relocatable build / install dir

### PR DESCRIPTION
**Pull request by benoit-pierre**
Pendent of: https://github.com/koreader/koreader-base/pull/1781

Include bumping base and some additional cleanups, most notably: move the platform update rules to dedicated files.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/11798)

### CODE REVIEW

**Comment by benoit-pierre** [comment]
This will conflict with #9351.

File: Makefile [hunk 1]
```diff
@@ -3,13 +3,13 @@ KOR_BASE?=base
 
 # the repository might not have been checked out yet, so make this
 # able to fail:
--include $(KOR_BASE)/Makefile.defs
+include $(KOR_BASE)/Makefile.defs
```
Comment (Frenzie) [inline Makefile:6]: Does the comment still make sense?

Comment (benoit-pierre) [reply Makefile:6]: Nope! Will amend.

**Comment by Frenzie** [comment]
> This will conflict with #9351.

Which one to merge first? Or put another way: how much?

**Comment by benoit-pierre** [comment]
Nothing major, just in `Makefile` because of the change to update rules. I've already resolved it locally. Smaller base bumps are better, IMHO, so I would merge this first.

**Review by Frenzie — APPROVED** [review]
lgtm

**Comment by Frenzie** [comment]
There seems to be a minor issue related to `ln -f`, even after removing the `koreader-emulator-x86_64-linux-gnu-debug` folder.
```
[*] Install plugins
ln -snfr /home/frans/src/ko/koreader/plugins koreader-emulator-x86_64-linux-gnu-debug/koreader/
[*] Install resources
ln -snfr /home/frans/src/ko/koreader/resources/fonts/* koreader-emulator-x86_64-linux-gnu-debug/koreader/fonts/
ln: koreader-emulator-x86_64-linux-gnu-debug/koreader/fonts/droid: cannot overwrite directory
ln: koreader-emulator-x86_64-linux-gnu-debug/koreader/fonts/freefont: cannot overwrite directory
ln: koreader-emulator-x86_64-linux-gnu-debug/koreader/fonts/nerdfonts: cannot overwrite directory
ln: koreader-emulator-x86_64-linux-gnu-debug/koreader/fonts/noto: cannot overwrite directory
ln: koreader-emulator-x86_64-linux-gnu-debug/koreader/fonts/urw: cannot overwrite directory
make: *** [Makefile:88: all] Error 1
Failed to build emulator! Try run with -v for more information.
```

**Comment by benoit-pierre** [comment]
I imagine this was with an old build directory: try removing `base/build/x86_64-pc-linux-gnu-debug/fonts` instead.

**Comment by Frenzie** [comment]
I thought I had, but you're probably right.

**Pull request merged**